service/controller/v25/resource/tccp/update.go

```diff
@@ -226,81 +219,3 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	return StackState{}, nil
 }
-
-// Terminates the master instance of the cluster.
-//
-// To detect the old master instance we find the instance by its name and
-// the instance state "stopped". Within the upgrade process the master first
-// gets stopped and its volumes get detached. This function makes sure that
-// the stopped instance is also terminated.
-func (r *Resource) terminateOldMasterInstance(ctx context.Context, cr v1alpha1.AWSConfig) error {
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	instanceName := key.MasterInstanceName(cr)
-
-	var instanceID string
-	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding master instance ID for %#q", instanceName))
-
-		i := &ec2.DescribeInstancesInput{
-			Filters: []*ec2.Filter{
-				{
-					Name: aws.String("tag:Name"),
-					Values: []*string{
-						aws.String(instanceName),
-					},
-				},
-				{
-					Name: aws.String("instance-state-name"),
-					Values: []*string{
-						aws.String("stopped"),
-					},
```

Comment:
Are you sure this isn't important during the update? I can imagine some EBS corruption because of that.

Reply:
What is the scenario though? The code before would mean we do not terminate actively and let cloud formation do whatever it does. And the master instance is replaced. One way or the other. And we do not relay the update to detach volumes again.

Reply:
Are EBS volumes detached during updated before `terminateOldMasterInstance` is called?

Reply:
I meant `terminateMasterInstance`.

Reply:
Let me look that up for you: :trollface: This here is in `master`: https://github.com/giantswarm/aws-operator/blob/master/service/controller/v25/resource/tccp/update.go#L50-L75.

Reply:
Okay, this seems to be fine then.

Reply:
Seems to work on `ginger`. The logs are "fine". They also expose the ebs service implementation weirdness. The master instance is stopped for each volume we detach. Not critical but also not super cool. 

```
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/update.go:45","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"updating the tenant cluster main stack","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:21.116197+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:84","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"requesting to stop instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:21.341448+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:97","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"requested to stop instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:21.600364+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:101","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"waiting for instance `i-06980d67d66f9428a` to stop","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:21.600404+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:114","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"waited for instance `i-06980d67d66f9428a` to stop","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:51.955402+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:118","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"detaching EBS volume `vol-0632b5bad4a0224a3` from instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:51.955444+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:132","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"detached EBS volume `vol-0632b5bad4a0224a3` from instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.242824+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:84","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"requesting to stop instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.24287+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:97","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"requested to stop instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.483327+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:101","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"waiting for instance `i-06980d67d66f9428a` to stop","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.483369+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:114","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"waited for instance `i-06980d67d66f9428a` to stop","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.532684+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:118","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"detaching EBS volume `vol-05e3b17a9268c843b` from instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.532742+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/ebs/ebs.go:132","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"detached EBS volume `vol-05e3b17a9268c843b` from instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.825458+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:24","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"finding master instance ID for `5s08r-master`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.825507+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:57","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"found master instance ID `i-06980d67d66f9428a` for `5s08r-master`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.986111+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:75","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"disabling termination protection for master instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:52.986157+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:89","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"disabled termination protection for master instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:53.102142+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:93","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"terminating master instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:53.102198+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/master_instance.go:106","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"terminated master instance `i-06980d67d66f9428a`","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:53.265901+00:00","version":"108732333"}
{"caller":"github.com/giantswarm/aws-operator/service/controller/v25/resource/tccp/update.go:106","controller":"aws-operator","event":"update","function":"ApplyUpdateChange","level":"debug","loop":"22","message":"updated the tenant cluster main stack","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/awsconfigs/5s08r","resource":"tccpv25","time":"2019-03-19T17:20:53.672257+00:00","version":"108732333"}